### PR TITLE
Support pagination for stream api

### PR DESCRIPTION
**Pull request by Huachao**

### CODE REVIEW

**Comment by Huachao** [comment]
Ping @duiniuluantanqin @winlinvip , could you please help to review?

**Review by duiniuluantanqin — APPROVED** [review]
the same as dumps_clients.

File: trunk/src/app/srs_app_statistic.cpp [hunk 1]
```diff
@@ -495,12 +495,16 @@ srs_error_t SrsStatistic::dumps_vhosts(SrsJsonArray* arr)
...
-srs_error_t SrsStatistic::dumps_streams(SrsJsonArray* arr)
+srs_error_t SrsStatistic::dumps_streams(SrsJsonArray* arr, int start, int count)
 {
     srs_error_t err = srs_success;
-    
-    std::map<std::string, SrsStatisticStream*>::iterator it;
-    for (it = streams.begin(); it != streams.end(); it++) {
+
+    std::map<std::string, SrsStatisticStream*>::iterator it = streams.begin();
+    for (int i = 0; i < start + count && it != streams.end(); i++) {
```
Comment (akanchi) [inline trunk/src/app/srs_app_statistic.cpp:503]: Iterator should be plus 1

Comment (Huachao) [reply trunk/src/app/srs_app_statistic.cpp:503]: nice catch, I will create another PR to fix

Comment (duiniuluantanqin) [reply trunk/src/app/srs_app_statistic.cpp:503]: Oh sorry, my careless.

Comment (Huachao) [reply trunk/src/app/srs_app_statistic.cpp:503]: My fault, I will create a PR ASAP

Comment (Huachao) [reply trunk/src/app/srs_app_statistic.cpp:503]: @akanchi @duiniuluantanqin I have created a new PR in https://github.com/ossrs/srs/pull/2582, please help to review and sorry for the inconvenience

**Pull request merged**